### PR TITLE
chore: delete share `merge`

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/1285

If CI passes, this seems safe to me

### CODE REVIEW

File: pkg/shares/share_merging.go [hunk 1]
```diff
@@ -6,64 +6,9 @@ import (
```
Comment (cmwaters) [inline pkg/shares/share_merging.go:6]: Should we rename this file?

File: pkg/shares/shares_test.go [hunk 2]
```diff
@@ -90,21 +38,6 @@ func TestPadFirstIndexedBlob(t *testing.T) {
...
-		case <-ctx.Done():
-			return
-		default:
-			TestMerge(t)
-		}
-	}
-}
-
 // generateRandomBlockData returns randomly generated block data for testing purposes
 func generateRandomBlockData(txCount, blobCount, maxSize int) (data coretypes.Data) {
```
Comment (cmwaters) [inline pkg/shares/shares_test.go:42]: Looks like this is also unused now

**Review by cmwaters — COMMENTED** [review]
Need to appease the linter but otherwise looks good

**Review by evan-forbes — CHANGES_REQUESTED** [review]
cross posting https://github.com/celestiaorg/celestia-app/issues/1285#issuecomment-1403500853

> merge doesn't appear used anywhere outside of shares_test and it isn't exported so it seems safe to delete entirely

I feel like we've discussed this a few times before but haven't dug up our past decisions. IIRC, while its not currently used in core/app, we do want to preserve this functionality because being able to go from the square to block data will be needed in the future, specicically for different types of partial nodes, and we should already have all of the pieces to do that for each reserved namespaces and blobs, so ideally there shouldn't be that much new necessary logic.

While the current roundtrip unit test/fuzzer for merging has not been maintained, it is a very convenient way to fuzz the share encoding. We create random block data, split it, then merge it and compare the input and output for equality.

I can see some reasoning for removing it, but I don't think us not using something now is water tight in this very specific scenario. If we do want to remove it, then I think we need to write down why we are removing it and in what cases we would add it. Not wanting to maintain the round trip test, or provide any guarantees that we even can do so, until we need the functionality is a perfectly good reason imo. We just need to agree that is the reasoning, and document that decision.

**Comment by rootulp** [comment]
I synced w/ @evan-forbes offline:
- If we maintain the `merge` function, it should be fixed, tested, and publicly exported
- If we don't want to maintain `merge` function, remove it and associated utils. 

We're leaning towards preserving it at the moment because the fuzz tester that uses it is still helpful.

**Pull request closed without merging**